src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -287,7 +287,10 @@ export class CellModel extends Disposable implements ICellModel {
 
 	public set hover(value: boolean) {
 		this._hover = value;
-		this.fireExecutionStateChanged();
+		// Skip changing the execution state while running to prevent focus changes from breaking things like user input
+		if (!this.isRunning) {
```

Comment:
Did you try removing this event firing completely? I'm not sure why we would ever need to fire the event for this - the state isn't actually changing.

Reply:
The execution state is used to control the class state of the RunCell button: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts#L121

The state gets set from the executionState event here: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts#L153

If you skip firing the event, then the Run button doesn't show up when hovering over an inactive cell.

Reply:
Sigh. This really shouldn't be named "cell execution state". Having the cell execution state be "hidden" makes no sense. 

Welp, probably out of scope for this. But it'd be great to look into at least updating the name of this thing so it's clearer that it's more than just the current state of the cell execution. Maybe `CellExecutionIconState`? Not sure if this is used for anything else than the icon though.

Reply:
Looks like CellExecutionState used in a few other places, so it isn't just for the icon state.

Reply:
😭